### DISCUSSION TITLE
Capstone DA - Flask UI (Error saat menjalankan applikasi web dashboard 'python app.py')

### DISCUSSION THREAD

**Discussion by MelissaRusli** [teamalgoritma/community#760] [Q&A] [answered]
Saya mengikuti instruksi pada video untuk menjalankan applikasi web dashboard tetapi mendapatkan kendala seperti berikut ketika saya meng-click link http yg muncul pada terminal:

<img width="973" alt="Screen Shot 2023-09-03 at 10 24 10 PM" src="https://github.com/teamalgoritma/community/assets/141122159/166c9d48-e154-46f0-ba6f-1e309a6236f4">

Dimana kemudian saya cek pada terminal window dan ada error sbb :

<img width="1187" alt="Screen Shot 2023-09-03 at 10 37 17 PM" src="https://github.com/teamalgoritma/community/assets/141122159/a20a0fd8-69cf-40f4-8397-941b7da705fb">

<img width="1254" alt="Screen Shot 2023-09-03 at 10 23 30 PM" src="https://github.com/teamalgoritma/community/assets/141122159/b25096c0-e597-4afc-98b3-c81bca3fdcf6">

<img width="906" alt="Screen Shot 2023-09-03 at 10 23 45 PM" src="https://github.com/teamalgoritma/community/assets/141122159/66131378-8001-4104-bc01-70f264f98a36">

Mohon dibantu kenapa bisa terjadi error? Saya juga menemukan thread yang mengalami hal yang serupa tetapi saya tidak menemukan info solusi nya pada thread tersebut.

**Comment by arkanayudhistira** [comment]
Dear @MelissaRusli,

Mengacu dari terminal yang ibu screenshot, sepertinya aplikasi sedang tidak berjalan (terdapat nama environment dan path di akhir). Apakah ibu menggunakan Ctrl+C untuk mengcopy link nya? karena menekan Ctrl+C akan menghentikan aplikasi tersebut sehingga tidak bisa dibuka. Untuk menjalankan link nya dapat dengan cara Ctrl+klik.

Semoga jawaban saya membantu!

**Reply by MelissaRusli** [reply]
Malam, saya menggunakan cmd+klik pada mouse (laptop saya macbook)

**Reply by MelissaRusli** [reply]
Saya mencoba kembali dengan membuka terminal baru, dan mengulangi proses yang sama dan kali ini hanya meng-klik link yang muncul pada tulisan biru 'Follow link' dan tetap tidak berhasil juga. Mohon bantuannya.

**Comment by arkanayudhistira** [comment]
Dear @MelissaRusli,

Mengacu dari diskusi pada forum berikut: https://stackoverflow.com/questions/63412583/nswindow-drag-regions-should-only-be-invalidated-on-the-main-thread-this-will-t , terdapat solusi yang dapat dicoba untuk mengatasi error tersebut yaitu menambahkan `matplotlib.pyplot.switch_backend('Agg') ` pada awal code app.py ibu. Silahkan dicoba terlebih dahulu

Semoga jawaban saya membantu!

**Reply by MelissaRusli** [reply, marked as answer]
Sudah saya coba tambahkan dan memberikan error sbb :

<img width="1090" alt="Screen Shot 2023-09-04 at 10 12 06 AM" src="https://github.com/teamalgoritma/community/assets/141122159/a27cb23f-76e9-4fe9-a786-11cf0db35070">

**Reply by MelissaRusli** [reply]
Sudah berhasil Mas Arkana, Terima Kasih bantuannya. Saya harus ganti matplotlib.pyplot ke variable plt dan berhasil.

**Reply by arkanayudhistira** [reply]
Baik, sama-sama ibu
